### PR TITLE
[DRAFT] New "Calliphorid" mutation tree - become a carrion eating flyperson.

### PR DESCRIPTION
**Pull request by Daved27hundred**
#### Summary
Content "Adds a new Fly mutation tree, a cross between insect and rat."

#### Purpose of change

New mutation lines are always nice, I've wanted something that makes use of the Eater of the Dead mutation from the rat tree.
The Calliphorid has the ability to eat rotten, mutated and even zombified corpses like the rat, but instead of having the utility of stainless claws and burrowing, the Calliphorid has enhanced perception and mobility, along with a small amount of armor.

Following Venera's suggestion, I'm making this a subcategory of the insect line instead of its own thing.

This is still a work in progress and is very likely to change.

#### Describe the solution

What's done:
- Added new mutations
- Added `SUB_CATEGORY` mutation category 

Todo:
- Add the threshold mutation and following mutations
- bugfix
- Anything I forgot to list here

#### Describe alternatives you've considered

I very much appreciate any feedback, recommendations or criticisms.

#### Testing

I plan to start testing this as soon as I have something of substance to test.

#### Additional Context

No I didn't make this PR just to make the joke in the branch name

### CODE REVIEW

**Comment by Daved27hundred** [comment]
of course I forget to lint it.

File: data/json/items/comestibles/mutagen.json [hunk 1]
```diff
@@ -217,6 +217,21 @@
       "vitamins": [ [ "mutagen_insect", 450, 550 ] ]
     }
   },
+  {
+    "id": "iv_mutagen_calliphorid",
+    "copy-from": "iv_mutagen_flavor",
+    "type": "COMESTIBLE",
+    "name": { "str": "calliphorid mutagenic primer" },
+    "description": "A processed mutagenic primer resembling blackened, rotted fat",
```
Comment (TheShadowFerret) [inline data/json/items/comestibles/mutagen.json:225]:
```suggestion
    "description": "A processed mutagenic primer resembling blackened, rotted fat.",
```

File: data/json/items/mutagen_ingredients.json [hunk 2]
```diff
@@ -144,6 +144,15 @@
     "looks_like": "medical_sample",
     "material": [ "iflesh", "chitin" ]
   },
+  {
+    "type": "GENERIC",
+    "id": "calliphorid_sample",
+    "copy-from": "mutagen_sample",
+    "color": "black",
+    "name": { "str": "fly sample" },
+    "description": "A slurry of decay",
```
Comment (TheShadowFerret) [inline data/json/items/mutagen_ingredients.json:153]:
```suggestion
    "description": "A slurry of decay.",
```

**Review by TheShadowFerret — CHANGES_REQUESTED** [review]
> No I didn't make this PR just to make the joke in the branch name

I don't believe you :3

**Comment by Venera3** [comment]
I'm not terribly enthusiastic about turning every single family into its own category. This would work just as well, be more easily attainable, and keep mutations from being 100% deterministic as an Insect subline. Something like
```
[ "CATEGORY_SUBTYPE" ] mutation category to make sure you're only on one subpath at any one time
"Calliphorid/Hymenopteran/Lepidopteran/Whatever Affinity" subfategory traits as a prereq for suitably fly-y/wasp-y/butterfly-y traits, with a second non-purifiable posttresh version to lock you into a phenotype for good
Some high-level traits sprinkled on top for those who go all the way
```
For Fly the Intestinal Fortitude and the basic Insect Wings line sounds right, but I'm drawing a blank on potential bad muts.

**Comment by Venera3** [comment]
To keep a limited scope and restrict it to fly stuff:
1) Make a new mutation type `SUB_CATEGORY` in mutation_types.json
2) Make a `Fly Affinity` trait with the `SUB_CATEGORY` type, dependent on one of the chitin traits and maybe the immature insect limbs - it doesn't need to have any effects on its own, this is your lower "threshold"
3) Have Fly-y traits need `Fly Affinity` (depending on which non-Insect trait you want to use you might need some duplicates) 
4) (extra credit) Add a nonpurifiable `Phly Phenotype` (requires affinity and the Fly traits) to lock you into your new flyperson body and/or unlock some highly flyy traits

**Comment by fairyarmadillo** [comment]
How about an acid barf attack using the spell system? Flies do that to their food, and so did Dr. Brundle. It could splash an adjacent target and make a puddle on the ground, hitting the player with the nausea effect for a bit afterwards.

Flies are pretty ugly and people don't like them, maybe they could pick up deformed or badly deformed as downsides. You could also give them their own version of the proboscis mutation, as house and carrion fly mouthparts are adapted for slurping up food they liquify by regurgitating digestive enzymes on it, but this might be too annoying since the player can't make meat soup this way, and would have to make protein shakes or something.

They could also get fast reflexes, 'cause flies are really good at dodging.

**Comment by Daved27hundred** [comment]
@fairyarmadillo  the improved dodge and the eating is part of what I plan to do. I didn't intend to make the proboscis upgrade prevent you from eating solids: You'd just take much longer to eat, because you'd need to vomit on the pemmican first.

As for the acid thing, I'm not entirely sure. I wanted to balance this mutation line by not giving it any natural weapons (claws, teeth and whatnot) like other ones. I'll think about it after I do what Venera said.

**Comment by Daved27hundred** [comment]
@Venera3 should I remove the `THRESH_CALLIPHORID` along with the mutagen recipes I wrote for it? since I'm making this a subcategory.

**Comment by Daved27hundred** [comment]
It's already set to a draft, but I'll put it in the PR name as well.

**Comment by Venera3** [comment]
> @Venera3 should I remove the `THRESH_CALLIPHORID` along with the mutagen recipes I wrote for it? since I'm making this a subcategory.

Yes, they will stay under Insect. Longish-term I could see a thing where subcategories like this have their own vitamins (so you either luck into them randomly or use mutagen made from flies to weigh the scales in their favor), but that's just idle musing at this stage.

**Comment by Daved27hundred** [comment]
Having an "issue" with putting saprovore + eater of the dead into this subcategory, I say that in quotations because I didn't actually try yet.
I thought I'd add `INSECT` to the categories they're available for and put `SUB_CALLIPHORID` as a prerequisite (so that a non-fly insect doesn't get it) but I'm worried that that would mess with the other mutation trees that use it. am I overthinking this?

**Comment by Venera3** [comment]
I'm pretty sure extra categories won't impact the base traits, though you want to test that the original categories are capable of gaining them still.

**Pull request closed without merging**